windows/ReactNativeWebView/ReactWebViewManager.cpp

```diff
@@ -75,7 +78,52 @@ namespace winrt::ReactNativeWebView::implementation {
                         uriString.replace(0, 7, bundleRootPath.empty() ? "ms-appx-web:///Bundle/" : bundleRootPath);
                     }
 
-                    webView.Navigate(winrt::Uri(to_hstring(uriString)));
+                    bool hasHeaders = srcMap.find("headers") != srcMap.end();
+                    auto httpRequest = winrt::HttpRequestMessage();
+                    httpRequest.RequestUri(winrt::Uri(to_hstring(uriString)));
+                    if (srcMap.find("method") != srcMap.end() && srcMap.at("method").AsString() == "POST")
+                    {
+                        httpRequest.Method(winrt::HttpMethod::Post());
+                        auto formBody = srcMap.at("body").AsString();
+                        bool isUrlEncodedForm = hasHeaders &&
+                            srcMap.at("headers").AsObject().find("content-type") !=
+                            srcMap.at("headers").AsObject().end() &&
+                            srcMap.at("headers").AsObject().at("content-type") == "application/x-www-form-urlencoded";
+                        if (isUrlEncodedForm)
+                        {
+                            auto formContent = winrt::single_threaded_observable_map<winrt::hstring, winrt::hstring>();
+                            auto counter = 0;
+                            auto current = formBody.find_first_of("&");
+                            while (counter <= formBody.find_last_of("&"))
+                            {
+                                auto keyValueSeparator = formBody.substr(counter, counter + current).find('=');
+                                if (keyValueSeparator <= current)
+                                {
+                                    auto key = winrt::to_hstring(formBody.substr(counter, keyValueSeparator));
+                                    auto value = winrt::to_hstring(formBody.substr(
+                                        keyValueSeparator + counter + 1, current - keyValueSeparator - 1));
+                                    formContent.Insert(key, value);
+                                }
+                                counter += current + 1;
+                                current = formBody.substr(counter, formBody.size() - counter).find_first_of("&");
+                            }
+                            httpRequest.Content(winrt::HttpFormUrlEncodedContent(formContent));
```

Comment:
Hey @jfkm69! thanks for this PR. This part seems error prone (there will be edge cases that would be good to not have to deal with) and like it might be trying to replicate the functionality in [Uri.QueryParsed](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.uri.queryparsed?view=winrt-19041); is there additional functionality you need that QueryParsed doesn't give you?
cc @chiaramooney